Userland/Libraries/LibGUI/WindowServerConnection.cpp

```diff
@@ -245,8 +245,23 @@ static MouseButton to_mouse_button(u32 button)
 
 void WindowServerConnection::mouse_down(i32 window_id, Gfx::IntPoint const& mouse_position, u32 button, u32 buttons, u32 modifiers, i32 wheel_delta_x, i32 wheel_delta_y)
 {
-    if (auto* window = Window::from_window_id(window_id))
-        Core::EventLoop::current().post_event(*window, make<MouseEvent>(Event::MouseDown, mouse_position, buttons, to_mouse_button(button), modifiers, wheel_delta_x, wheel_delta_y));
+    auto* window = Window::from_window_id(window_id);
+    if (!window)
+        return;
+
+    auto mouse_event = make<MouseEvent>(Event::MouseDown, mouse_position, buttons, to_mouse_button(button), modifiers, wheel_delta_x, wheel_delta_y);
+
+    if (auto* action = action_for_shortcut(*window, Shortcut(mouse_event->modifiers(), mouse_event->button()))) {
+        if (action->is_enabled()) {
+            action->flash_menubar_menu(*window);
+            action->activate();
+            return;
+        }
+        if (action->swallow_key_event_when_disabled())
```

Comment:
Does it make sense to check for this given this will always be a "Mouse Event" and never a "Key Event"?

Reply:
Yeah I wasn't sure if I should update that method name to not mention _key_ events specifically, eg. `swallow_input_event_when_disabled()`, which would then allow it to be sensibly called from both key- and mouse-down, or if should just not call for mouse-down. 

Did you have any thoughts on best approach?